wrapper.js

```diff
@@ -14,8 +14,9 @@ function setupMethods (soljson) {
   var compileStandard = null;
   if (('_compileJSONCallback' in soljson) || ('_compileStandard' in soljson)) {
     var copyString = function (str, ptr) {
-      var buffer = soljson._malloc(str.length + 1);
-      soljson.writeStringToMemory(str, buffer);
+      var length = soljson.lengthBytesUTF8(str);
+      var buffer = soljson._malloc(length + 1);
+      soljson.stringToUTF8(str, buffer, length + 1);
```

Comment:
Code here: https://github.com/kripken/emscripten/blob/78b44ed55cc4d0b4d79f62df9e80ae6f29a5345b/src/preamble.js#L1612-L1625

> // stringToUTF8Array always appends null.

> // Copies the given Javascript String object 'str' to the emscripten HEAP at address 'outPtr',
> // null-terminated and encoded in UTF8 form. The copy will require at most str.length*4+1 bytes of space in the HEAP.
> // Use the function lengthBytesUTF8 to compute the exact number of bytes (excluding null terminator) that this function will write.
> // Returns the number of bytes written, EXCLUDING the null terminator.

So I think the above code now is correct.